crates/forge_main/src/main.rs

```diff
@@ -26,17 +25,7 @@ async fn main() -> Result<()> {
     }));
 
     // Initialize and run the UI
-    let mut cli = Cli::parse();
-
-    // Check if there's piped input and no explicit prompt was provided
-    if cli.prompt.is_none() && !atty::is(atty::Stream::Stdin) {
-        let mut stdin_content = String::new();
-        std::io::stdin().read_to_string(&mut stdin_content)?;
-        let trimmed_content = stdin_content.trim();
-        if !trimmed_content.is_empty() {
-            cli.prompt = Some(trimmed_content.to_string());
-        }
```

Comment:
I think the bug was around us not setting the prompt when trim is empty. It works if we remove this condition.